features/daily-image.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const axios = require('axios');
+
+module.exports = function(controller) {
+ 
+  controller.on('slash_command', async (bot, message) => {
+    console.log(message);
+    if (message.command == '/daily-image') {
+      let imageSearchString = getImageQueryString(message.text.trim());
+      console.log('Image Search String = '+imageSearchString);
+      //This is an interesting workaround to handle timing of the nested promises
+      await bot.changeContext(message.reference);
+      await getImage(bot, message, imageSearchString);
+    }
+  });
+ };
+
+async function getImage(bot, message, imageSearchString) {
+  let searchUrl = 'https://api.bing.microsoft.com/v7.0/images/search?q='+imageSearchString+'&size=medium&mkt=en-us&count=50'
+  await axios.get(searchUrl, { 
```

Comment:
It seems to me like this should require an await _and_ a .then() block. Is there a reason to use both of them here?

Reply:
Ah, I think it is redundant. Good catch.